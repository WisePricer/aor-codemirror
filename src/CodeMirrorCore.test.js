import { CodeMirrorCore } from './CodeMirrorCore'
import { Controlled } from 'react-codemirror2'
import { Button } from '@material-ui/core'

const defaultProps = {
  label: 'Field Label',
  type: 'json',
  meta: {
    valid: true
  }
}

describe('CodeMirrorCore', () => {
  it('renders with defaults', () => {
    expect(shallow(<CodeMirrorCore {...defaultProps} />)).toMatchSnapshot()
  })

  it('renders with errors', () => {
    expect(shallow(<CodeMirrorCore {...defaultProps} meta={{ valid: false, error: 'Field error!' }} />)).toMatchSnapshot()
  })

  it('renders controlled', () => {
    const value = 'original value'
    const onChange = jest.fn()
    const wrapper = shallow(<CodeMirrorCore {...defaultProps} input={{ value, onChange }} />)
    expect(wrapper).toMatchSnapshot()

    const newValue = 'new value'
    wrapper.find(Controlled).simulate('beforeChange', null, null, newValue)
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toHaveBeenCalledWith(newValue)
  })

  it('formats value', () => {
    const onChange = jest.fn()
    const wrapper = shallow(<CodeMirrorCore {...defaultProps} input={{ value: '{"foo":"bar"}', onChange }} />)

    wrapper.find(Button).simulate('click')
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toHaveBeenCalledWith(`{
  "foo": "bar"
}`)
  })
})
