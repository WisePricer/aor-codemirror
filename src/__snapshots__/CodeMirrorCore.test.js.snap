// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeMirrorCore renders controlled 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#c4c4c4",
      }
    }
  >
    Field Label
     
  </span>
  <WithStyles(ForwardRef(Toolbar))
    disableGutters={true}
  >
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      Format
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Toolbar))>
  <Controlled
    label="Field Label"
    onBeforeChange={[Function]}
    options={
      Object {
        "lineNumbers": true,
        "lineWrapping": true,
        "mode": "application/json",
      }
    }
    value="original value"
  />
</div>
`;

exports[`CodeMirrorCore renders with defaults 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#c4c4c4",
      }
    }
  >
    Field Label
     
  </span>
  <WithStyles(ForwardRef(Toolbar))
    disableGutters={true}
  >
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      Format
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Toolbar))>
  <Controlled
    label="Field Label"
    onBeforeChange={[Function]}
    options={
      Object {
        "lineNumbers": true,
        "lineWrapping": true,
        "mode": "application/json",
      }
    }
    value=""
  />
</div>
`;

exports[`CodeMirrorCore renders with errors 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#c4c4c4",
      }
    }
  >
    Field Label
     
  </span>
  <span
    style={
      Object {
        "color": "#ff3d3a",
      }
    }
  >
     (
    Field error!
    )
  </span>
  <WithStyles(ForwardRef(Toolbar))
    disableGutters={true}
  >
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      Format
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Toolbar))>
  <Controlled
    label="Field Label"
    onBeforeChange={[Function]}
    options={
      Object {
        "lineNumbers": true,
        "lineWrapping": true,
        "mode": "application/json",
      }
    }
    value=""
  />
</div>
`;
